lib/active_model/serializer.rb

```diff
@@ -199,16 +199,22 @@ def serializer_from_options(options)
       opts
     end
 
+    def self.serializers_cache
+      @serializers_cache ||= ThreadSafe::Cache.new
```

Comment:
I share @joaomdmoura's concern about low memory servers (being as I'm on a `1x` Heroku dyno). How much memory are we talking?

Yet another reason why #832 is a great idea.